cfgov/jinja2/v1/_includes/organisms/fig-section.html

```diff
@@ -0,0 +1,25 @@
+<div>
+    {% if block.block_type == 'Fig_Section' %}
+        <h2 class="report-header">
+	    <a id="{{ value.section_id }}" 
```

Comment:
Could these IDs be generated dynamically, as part of the rendering, or is it important that they live in the page data? If they don't have to actually persist in the database there'd be no need to do the page-editing complications above. (I do note that the previous implementation you change here worked more like that, so forgive me if this suggestion isn't helpful.)

One approach to generate at render time (even with the StreamField-based approach) would be to create some kind of Jinja tag that returned an appropriate ID each time it was called, so if you had a template (or multiple nested templates) with calls like this:

```html
{{ get_section_number('section') }}
{{ get_section_number('subsection') }}
{{ get_section_number('section') }}
{{ get_section_number('subsection') }}
```

it'd render like this:

```
1.
1.1.
2.
2.1.
```

We have an example of doing something like that [here](https://github.com/cfpb/consumerfinance.gov/blob/54dddc061b5f843a575a6887bb412a4033de8365/cfgov/v1/jinja2tags/__init__.py#L99); this works by storing the current state as you render onto the request object.

Reply:
I hadn't considered that kind of approach at all, super interesting! I'm still not sure which approach is better (generating section IDs at save time vs page load time), but here was my reasoning for assigning them at save time:
1. A content editor might benefit from seeing the section IDs in the Wagtail admin, to help them orient when editing the content. (Just speculation -- I haven't tested this)
2. The section IDs in the table of contents need to match the section IDs in the main page content, since the anchor links are based on section ID. I figured if we calculate the IDs at page load time, we'd be doing so twice -- once for TOC, and again for main content, which could be more error-prone. (Again, speculation)

But I don't love the solution I've made either. It has section ID as editable field when the user isn't supposed to edit it, which is annoying. It also has the issue you noted above where the section IDs don't get assigned before previewing, though I'll look into the solution you suggested.